pull_buildpack.go

```diff
@@ -0,0 +1,80 @@
+package pack
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/config"
+	"github.com/buildpacks/pack/internal/buildpack"
+	"github.com/buildpacks/pack/internal/buildpackage"
+	"github.com/buildpacks/pack/internal/dist"
+	"github.com/buildpacks/pack/internal/style"
+)
+
+type PullBuildpackOptions struct {
+	URI          string
+	RegistryType string
+	RegistryURL  string
+	RegistryName string
+}
+
+func (c *Client) PullBuildpack(ctx context.Context, opts PullBuildpackOptions) error {
+	locatorType, err := buildpack.GetLocatorType(opts.URI, []dist.BuildpackInfo{})
+	if err != nil {
+		return err
+	}
+
+	switch locatorType {
+	case buildpack.URILocator:
```

Comment:
If I'm not mistaken this code path would download the files, for example `.cnb`, but wouldn't actually load them into the daemon.

Also, some tests that prove that the docker daemon has the images would be useful.

Reply:
Actually, I realize now that we don't need to support `URILocator` in this command (pullPolicy doesn't apply to it, so there's not much of a use case for needing it). Give that saving the blob to the daemon would be a fairly new chunk of code, I'm going to omit this.